pulsar-common/src/main/java/org/apache/pulsar/common/api/DoubleByteBuf.java

```diff
@@ -101,385 +65,50 @@ public ByteBuf getSecond() {
-    public int setBytes(int index, ScatteringByteChannel in, int length) throws IOException {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected short _getShortLE(int index) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected int _getUnsignedMediumLE(int index) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected int _getIntLE(int index) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected long _getLongLE(int index) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected void _setShortLE(int index, int value) {
-        throw new UnsupportedOperationException();
+    public int readableBytes() {
+        return b1.readableBytes() + b2.readableBytes();
     }
 
-    @Override
-    protected void _setMediumLE(int index, int value) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected void _setIntLE(int index, int value) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    protected void _setLongLE(int index, long value) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public int getBytes(int index, FileChannel out, long position, int length) throws IOException {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public int setBytes(int index, FileChannel in, long position, int length) throws IOException {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public ByteBuf copy(int index, int length) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public int nioBufferCount() {
-        return b1.nioBufferCount() + b2.nioBufferCount();
-    }
-
-    @Override
-    public ByteBuffer internalNioBuffer(int index, int length) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public ByteBuffer nioBuffer(int index, int length) {
-        ByteBuffer dst = ByteBuffer.allocate(length);
-        ByteBuf b = Unpooled.wrappedBuffer(dst);
-        b.writerIndex(0);
-        getBytes(index, b, length);
-        return dst;
-    }
-
-    @Override
-    public ByteBuffer[] nioBuffers(int index, int length) {
-        return new ByteBuffer[] { nioBuffer(index, length) };
-    }
-
-    @Override
-    public DoubleByteBuf discardReadBytes() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public String toString() {
-        String result = super.toString();
-        result = result.substring(0, result.length() - 1);
-        return result + ", components=2)";
-    }
-
-    @Override
-    public ByteBuffer[] nioBuffers() {
-        return nioBuffers(readerIndex(), readableBytes());
+    /**
+     * @return a single buffer with the content of both individual buffers
+     */
+    public ByteBuf coalesce() {
```

Comment:
I would remove this method un favor of a static method for tests

Reply:
Good point. I'll also expand the doc to mention it.